code/modules/surgery/robotics.dm

```diff
@@ -3,581 +3,646 @@
-	"<span class='notice'> You close and secure the hatch on [target]'s [affected.name] with \the [tool].</span>")
-	affected.open = 0
-	return TRUE
+	user.visible_message(
+		"<span class='notice'> [user] closes and secures the hatch on [target]'s [affected.name] with \the [tool].</span>",
+		"<span class='notice'> You close and secure the hatch on [target]'s [affected.name] with \the [tool].</span>"
+	)
+	tool.play_tool_sound(target)
+	affected.open = ORGAN_CLOSED
+	return SURGERY_STEP_CONTINUE
 
-/datum/surgery_step/robotics/external/close_hatch/fail_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool,datum/surgery/surgery)
+/datum/surgery_step/robotics/external/close_hatch/fail_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool, datum/surgery/surgery)
 	var/obj/item/organ/external/affected = target.get_organ(target_zone)
 	user.visible_message("<span class='warning'> [user]'s [tool.name] slips, failing to close the hatch on [target]'s [affected.name].</span>",
 	"<span class='warning'> Your [tool.name] slips, failing to close the hatch on [target]'s [affected.name].</span>")
-	return FALSE
+	return SURGERY_STEP_RETRY
 
-/datum/surgery_step/robotics/external/repair
-	name = "repair damage internally"
-	allowed_tools = list()
+/datum/surgery_step/robotics/external/close_hatch/premature
+	name = "close hatch prematurely"
 
-	var/list/implements_finish = list(
-		/obj/item/retractor = 100,
-		/obj/item/crowbar = 100,
-		/obj/item/kitchen/utensil = 50
+/datum/surgery_step/robotics/external/close_hatch/premature/begin_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+	user.visible_message(
+		"[user] begins to close and secure the hatch on [target]'s [affected.name] with \the [tool].",
+		"<span class'warning'>You are interrupting the current surgery</span>, beginning to close and secure the hatch on [target]'s [affected.name] with \the [tool]."
 	)
-	var/list/implements_heal_burn = list(
+	return ..()
+
+
+/datum/surgery_step/robotics/external/repair
+	name = "repair damage"
+	time = 3.2 SECONDS
+
+/datum/surgery_step/robotics/external/repair/burn
+	name = "repair burn damage"
+	allowed_tools = list(
 		/obj/item/stack/cable_coil = 100
 	)
-	var/list/implements_heal_brute = list(
-		/obj/item/weldingtool = 100,
+
+/datum/surgery_step/robotics/external/repair/burn/begin_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+	var/obj/item/stack/cable_coil/C = tool
+	if(!(affected.burn_dam > 0))
+		to_chat(user, "<span class='warning'>\The [affected] does not have any burn damage!</span>")
+		return SURGERY_BEGINSTEP_SKIP
+	if(!istype(C))
+		return SURGERY_BEGINSTEP_SKIP
+	if(!C.get_amount() >= 3)
```

Comment:
Think this should be 
```suggestion
	if(C.get_amount() < 3)
```
Because `!C.get_amount()` would just be 0 for any non-zero amount of cables.